instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/resources/SpringResourceConfigProperties.java

```diff
@@ -54,22 +66,44 @@ public Double getDouble(String name) {
     return environment.getProperty(name, Double.class);
   }
 
-  @Nullable
-  @Override
-  public Duration getDuration(String name) {
-    return environment.getProperty(name, Duration.class);
-  }
-
   @SuppressWarnings("unchecked")
   @Override
   public List<String> getList(String name) {
     return (List<String>) environment.getProperty(name, List.class);
   }
 
+  @Nullable
+  @Override
+  public Duration getDuration(String name) {
+    String value = getString(name);
+    if (value == null) {
+      return null;
+    }
+    return DefaultConfigProperties.createFromMap(Collections.singletonMap(name, value))
+        .getDuration(name);
+  }
+
   @SuppressWarnings("unchecked")
   @Override
   public Map<String, String> getMap(String name) {
+    // maps from config properties are not supported by Environment, so we have to fake it
```

Comment:
can the same approach used above in `getDuration` be used here?

Reply:
No - maps are not exposed via `environment` at all - spring recommends that you only use configuration properties for new applications.

Reply:
I see, so spring boot starter wouldn't support `OTEL_EXPORTER_OTLP_HEADERS` env var? I think that's fine, and aligns with https://github.com/open-telemetry/opentelemetry-specification/pull/3850

Reply:
cc @jack-berg (example of the benefit of https://github.com/open-telemetry/opentelemetry-specification/pull/3850)

Reply:
> OTEL_EXPORTER_OTLP_HEADERS

Yes, this also works:
- transforms a string to a map, because of [MapConverter](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/internal/MapConverter.java)
- maps env vars to spring properties
- we read [the property](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/10292/files#diff-4ea348bfafe6e9681ce8ea0747e71192e99c4489998c6089d11cd0d431b40cffR93)

Reply:
> config file considerations

I would like to get the spring starter in a good shape for GA before we look into supporting upcoming features